evaluator/evaluator.go

```diff
@@ -237,29 +237,33 @@ func evalInfixExpression(operator string, left object.Object, right object.Objec
 func evalIntegerInfixExpression(operator string, left object.Object, right object.Object) object.Object {
 	lVal := left.(*object.Integer).Value
 	rVal := right.(*object.Integer).Value
+	var res object.Object
 
 	switch operator {
 	case "+":
-		return &object.Integer{Value: lVal + rVal}
+		res = &object.Integer{Value: lVal + rVal}
 	case "-":
-		return &object.Integer{Value: lVal - rVal}
+		res = &object.Integer{Value: lVal - rVal}
 	case "*":
-		return &object.Integer{Value: lVal * rVal}
+		// TODO (Peter) if one number is actually zero we only need to depend on it!
```

Comment:
Tell me more here? Not sure I'm following

Reply:
For example consider:
```
x = a * b * c
```
As long as `a = 0` the values of `b,c` do not matter so they can be excluded as dependencies